napalm_eos/eos.py

```diff
@@ -1201,3 +1201,140 @@ def traceroute(self, destination, source='', ttl=0, timeout=0):
                 previous_probe_ip_address = ip_address
 
         return traceroute_result
+
+    def get_bgp_neighbors_detail(self, neighbor_address=''):
+        """Function to return detailed BGP information.
+
+        Information returned by this function can be about a single peer or
+        about all peers. This can be controlled using the following:
+
+        :params neighbor_address: Retuns the statistics for a specific
+                                  BGP neighbor or for all BGP neighbors.
+        """
+
+        commands = []
+        if not neighbor_address:
+            commands.append('show ip bgp neighbors vrf all')
+            commands.append('show ipv6 bgp neighbors vrf all')
+        else:
+            try:
+                peer_ver = IPAddress(neighbor_address).version
+            except Exception:
+                LOG.error('Invalid BGP neighbor address %s' % neighbor_address)
+                return
+
+            if peer_ver == 4:
+                commands.append('show ip bgp neighbors %s vrf all' %
+                                neighbor_address)
+            elif peer_ver == 6:
+                commands.append('show ipv6 bgp neighbors %s vrf all' %
+                                neighbor_address)
+
+        raw_output = (
+            self.device.run_commands(commands, encoding='text'))
+
+        bgp_detail_info = {}
+
+        if neighbor_address:
+            peer_info = self._parse_per_peer_bgp_detail(
```

Comment:
Hang on a second - the name of the method confused me a bit: ```_parse_per_peer_bgp_detail``` actually is called once, at most twice, right?